File: GitHub/Models/Team.cs
using Microsoft.Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
namespace GithubTodoDemo.GitHub.Models {
    /// <summary>Groups of organization members that gives permissions on specified repositories.</summary>
    public class Team : IAdditionalDataHolder, IParsable {
        /// <summary>Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.</summary>
        public IDictionary<string, object> AdditionalData { get; set; }
        /// <summary>The description property</summary>
        public string Description { get; set; }
        /// <summary>The html_url property</summary>
        public string Html_url { get; set; }
        /// <summary>The id property</summary>
        public int? Id { get; set; }
        /// <summary>The members_url property</summary>
        public string Members_url { get; set; }
        /// <summary>The name property</summary>
        public string Name { get; set; }
        /// <summary>The node_id property</summary>
        public string Node_id { get; set; }
        /// <summary>Groups of organization members that gives permissions on specified repositories.</summary>
        public NullableTeamSimple Parent { get; set; }
        /// <summary>The permission property</summary>
        public string Permission { get; set; }
        /// <summary>The permissions property</summary>
        public Team_permissions Permissions { get; set; }
        /// <summary>The privacy property</summary>
        public string Privacy { get; set; }
        /// <summary>The repositories_url property</summary>
        public string Repositories_url { get; set; }
        /// <summary>The slug property</summary>
        public string Slug { get; set; }
        /// <summary>The url property</summary>
        public string Url { get; set; }
        /// <summary>
        /// Instantiates a new team and sets the default values.
        /// </summary>
        public Team() {
            AdditionalData = new Dictionary<string, object>();
        }
        /// <summary>
        /// Creates a new instance of the appropriate class based on discriminator value
        /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
        /// </summary>
        public static Team CreateFromDiscriminatorValue(IParseNode parseNode) {
            _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
            return new Team();
        }
        /// <summary>
        /// The deserialization information for the current model
        /// </summary>
        public IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
            return new Dictionary<string, Action<IParseNode>> {
                {"description", n => { Description = n.GetStringValue(); } },
                {"html_url", n => { Html_url = n.GetStringValue(); } },
                {"id", n => { Id = n.GetIntValue(); } },
                {"members_url", n => { Members_url = n.GetStringValue(); } },
                {"name", n => { Name = n.GetStringValue(); } },
                {"node_id", n => { Node_id = n.GetStringValue(); } },
                {"parent", n => { Parent = n.GetObjectValue<NullableTeamSimple>(NullableTeamSimple.CreateFromDiscriminatorValue); } },
                {"permission", n => { Permission = n.GetStringValue(); } },
                {"permissions", n => { Permissions = n.GetObjectValue<Team_permissions>(Team_permissions.CreateFromDiscriminatorValue); } },
                {"privacy", n => { Privacy = n.GetStringValue(); } },
                {"repositories_url", n => { Repositories_url = n.GetStringValue(); } },
                {"slug", n => { Slug = n.GetStringValue(); } },
                {"url", n => { Url = n.GetStringValue(); } },
            };
        }
        /// <summary>
        /// Serializes information the current object
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        /// </summary>
        public void Serialize(ISerializationWriter writer) {
            _ = writer ?? throw new ArgumentNullException(nameof(writer));
            writer.WriteStringValue("description", Description);
            writer.WriteStringValue("html_url", Html_url);
            writer.WriteIntValue("id", Id);
            writer.WriteStringValue("members_url", Members_url);
            writer.WriteStringValue("name", Name);
            writer.WriteStringValue("node_id", Node_id);
            writer.WriteObjectValue<NullableTeamSimple>("parent", Parent);
            writer.WriteStringValue("permission", Permission);
            writer.WriteObjectValue<Team_permissions>("permissions", Permissions);
            writer.WriteStringValue("privacy", Privacy);
            writer.WriteStringValue("repositories_url", Repositories_url);
            writer.WriteStringValue("slug", Slug);
            writer.WriteStringValue("url", Url);
            writer.WriteAdditionalData(AdditionalData);
        }
    }
}

File: python/githubclient/models/team.py
from __future__ import annotations
from collections.abc import Callable
from dataclasses import dataclass, field
from kiota_abstractions.serialization import AdditionalDataHolder, Parsable, ParseNode, SerializationWriter
from typing import Any, Optional, TYPE_CHECKING, Union

if TYPE_CHECKING:
    from .nullable_team_simple import NullableTeamSimple
    from .team_permissions import Team_permissions

@dataclass
class Team(AdditionalDataHolder, Parsable):
    """
    Groups of organization members that gives permissions on specified repositories.
    """
    # Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
    additional_data: dict[str, Any] = field(default_factory=dict)

    # The description property
    description: Optional[str] = None
    # The html_url property
    html_url: Optional[str] = None
    # The id property
    id: Optional[int] = None
    # The members_url property
    members_url: Optional[str] = None
    # The name property
    name: Optional[str] = None
    # The node_id property
    node_id: Optional[str] = None
    # Groups of organization members that gives permissions on specified repositories.
    parent: Optional[NullableTeamSimple] = None
    # The permission property
    permission: Optional[str] = None
    # The permissions property
    permissions: Optional[Team_permissions] = None
    # The privacy property
    privacy: Optional[str] = None
    # The repositories_url property
    repositories_url: Optional[str] = None
    # The slug property
    slug: Optional[str] = None
    # The url property
    url: Optional[str] = None
    
    @staticmethod
    def create_from_discriminator_value(parse_node: ParseNode) -> Team:
        """
        Creates a new instance of the appropriate class based on discriminator value
        param parse_node: The parse node to use to read the discriminator value and create the object
        Returns: Team
        """
        if parse_node is None:
            raise TypeError("parse_node cannot be null.")
        return Team()
    
    def get_field_deserializers(self,) -> dict[str, Callable[[ParseNode], None]]:
        """
        The deserialization information for the current model
        Returns: dict[str, Callable[[ParseNode], None]]
        """
        from .nullable_team_simple import NullableTeamSimple
        from .team_permissions import Team_permissions

        from .nullable_team_simple import NullableTeamSimple
        from .team_permissions import Team_permissions

        fields: dict[str, Callable[[Any], None]] = {
            "description": lambda n : setattr(self, 'description', n.get_str_value()),
            "html_url": lambda n : setattr(self, 'html_url', n.get_str_value()),
            "id": lambda n : setattr(self, 'id', n.get_int_value()),
            "members_url": lambda n : setattr(self, 'members_url', n.get_str_value()),
            "name": lambda n : setattr(self, 'name', n.get_str_value()),
            "node_id": lambda n : setattr(self, 'node_id', n.get_str_value()),
            "parent": lambda n : setattr(self, 'parent', n.get_object_value(NullableTeamSimple)),
            "permission": lambda n : setattr(self, 'permission', n.get_str_value()),
            "permissions": lambda n : setattr(self, 'permissions', n.get_object_value(Team_permissions)),
            "privacy": lambda n : setattr(self, 'privacy', n.get_str_value()),
            "repositories_url": lambda n : setattr(self, 'repositories_url', n.get_str_value()),
            "slug": lambda n : setattr(self, 'slug', n.get_str_value()),
            "url": lambda n : setattr(self, 'url', n.get_str_value()),
        }
        return fields
    
    def serialize(self,writer: SerializationWriter) -> None:
        """
        Serializes information the current object
        param writer: Serialization writer to use to serialize this model
        Returns: None
        """
        if writer is None:
            raise TypeError("writer cannot be null.")
        writer.write_str_value("description", self.description)
        writer.write_str_value("html_url", self.html_url)
        writer.write_int_value("id", self.id)
        writer.write_str_value("members_url", self.members_url)
        writer.write_str_value("name", self.name)
        writer.write_str_value("node_id", self.node_id)
        writer.write_object_value("parent", self.parent)
        writer.write_str_value("permission", self.permission)
        writer.write_object_value("permissions", self.permissions)
        writer.write_str_value("privacy", self.privacy)
        writer.write_str_value("repositories_url", self.repositories_url)
        writer.write_str_value("slug", self.slug)
        writer.write_str_value("url", self.url)
        writer.write_additional_data_value(self.additional_data)
    

